corehq/apps/hqmedia/models.py

```diff
@@ -742,12 +746,19 @@ def remove_unused_mappings(self):
             return
         paths = list(self.multimedia_map) if self.multimedia_map else []
         permitted_paths = self.all_media_paths() | self.logo_paths
+        deleted_media = []
+        allow_deletion = self.domain not in {'icds-cas', 'icds-test'}
         for path in paths:
             if path not in permitted_paths:
-                map_changed = True
                 map_item = self.multimedia_map[path]
-                _log_media_deletion(self, map_item, path)
-                del self.multimedia_map[path]
+                deleted_media.append((path, map_item, None))
+                if not allow_deletion:
+                    map_changed = True
+                    del self.multimedia_map[path]
```

Comment:
This is deleting only `if not allow_deletion`, which is confusing. Is that what you intended? Now this will only delete if the domain is 'icds-cas' or 'icds-test'. Seems like possibly not what you wanted.

Reply:
yes, thanks. I swapped it around above but not here.

Reply:
updated